### PR TITLE
Feat: Refactoring and Various Enhancements

### PR DESCRIPTION
**Pull request by refcell**
## Overview

- [x] Refactors and upgrades clap cli argument parsing to stable.
- [x] Introduces testing.
~~- [ ] Packet Encoding.~~ (split into future pr)

### CODE REVIEW

File: bootstrap.json [hunk 1]
```diff
@@ -0,0 +1,2090 @@
+{
+    "data": [
```
Comment (AgeManning) [inline bootstrap.json:2]: Probably for the main repo, we don't want to store this kinda of data, which I assume refers to a specific network. Maybe we should just have a single entry (with a fake IP) to demonstrate the format?

File: Cargo.toml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 [package]
 name = "discv5-cli"
-version = "0.2.4"
-description = "Simple utility to test discv5"
+version = "0.2.5"
+description = "Simple CLI utility for creating and debugging discovery v5 servers"
 authors = ["Age Manning <Age@AgeManning.com>"]
```
Comment (AgeManning) [inline Cargo.toml:5]: Looks like you've pretty involved in this, feel free to add yourself.

File: Cargo.toml [hunk 3]
```diff
@@ -1,12 +1,12 @@
 [package]
 name = "discv5-cli"
-version = "0.2.4"
-description = "Simple utility to test discv5"
+version = "0.2.5"
```
Comment (AgeManning) [inline Cargo.toml:3]: The changes are substantial, what do you think about bumping a whole minor version?

**Review by AgeManning — COMMENTED** [review]
Awesome. Thanks for the contrib. Seems you're using this fairly extensively. 
Few minor comments.

File: bootstrap.json (hunk 1 above)
Comment (refcell) [reply bootstrap.json:2]: Agreed!

File: Cargo.toml (hunk 2 above)
Comment (refcell) [reply Cargo.toml:5]: Ah no worries, thanks though!

File: Cargo.toml (hunk 3 above)
Comment (refcell) [reply Cargo.toml:3]: SGTM

**Comment by refcell** [comment]
> Awesome. Thanks for the contrib. Seems you're using this fairly extensively. Few minor comments.

Thanks! Really just trying to learn so figured I'd update so others can learn more as well :)

**Review by AgeManning — APPROVED** [review]

**Pull request merged**